python/ray/data/_internal/planner/aggregate.py

```diff
@@ -73,6 +74,13 @@ def fn(
         else:
             scheduler = PullBasedShuffleTaskScheduler(agg_spec)
 
-        return scheduler.execute(refs, num_outputs, ctx)
+        return scheduler.execute(
+            refs,
+            num_outputs,
+            ctx,
+            _debug_limit_execution_to_num_blocks=(
```

Comment:
I think this is fine given that we already have the same code duplication for the other args, and it seems better to me to be explicit about what args `scheduler` should take, instead of passing implicitly through the current DataConfig.

Reply:
ok. I still have slightly stronger preference on using `DataContext` or `TaskContext`. Especially because the scheduler already has a "context" parameter, and this argument can be considered as part of the context. But I'll leave the decision to you.